volumetric/volMask.cpp

```diff
@@ -0,0 +1,255 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file
+ * @ingroup volumetric
+ * @author Bertrand Kerautret (\c bertrand.kerautret@univ-lyon2.fr )
+ * @author Jonas Lamy (\c jonas.lamy@univ-lyon2.fr )
+ *
+ *
+ * @date 2019/03/01
+ *
+ * Source file of the tool volMask
+ *
+ * This file is part of the DGtal library/DGtalTools Project.
+ */
+
+///////////////////////////////////////////////////////////////////////////////
+#include "DGtal/base/Common.h"
+#include <DGtal/helpers/StdDefs.h>
+#include <DGtal/images/ImageContainerBySTLVector.h>
+
+#include <DGtal/io/readers/GenericReader.h>
+#include <DGtal/io/writers/GenericWriter.h>
+#include <boost/program_options/options_description.hpp>
+#include <boost/program_options/parsers.hpp>
+#include <boost/program_options/variables_map.hpp>
+#ifdef WITH_ITK
+#include "DGtal/io/readers/ITKReader.h"
+#include "DGtal/io/writers/ITKWriter.h"
+
+#endif
+
+
+///////////////////////////////////////////////////////////////////////////////
+using namespace std;
+using namespace DGtal;
+///////////////////////////////////////////////////////////////////////////////
+namespace po = boost::program_options;
+
+
+/**
+ @page volMask volMask
+ 
+ @brief  Extracts a new image from the a mask image that represents the regions of the image which are selected and copied in the resulting image. Elements outside the regions defined by the mask are set to 0.
```

Comment:
mm.. I don't get it ;)
Is the mask binary ?

Reply:
You filter the input image by selecting the voxels with true (or maskValue) values in the mask ?
What would be the values "outside"?

Reply:
> Is the mask binary ?

Yes 

>You filter the input image by selecting the voxels with true (or maskValue) values in the mask ?

exactly

> What would be the values "outside"?

They are set to 0. Typically it is used it when we have an organ segmented  through a binary image (considered as the image mask) and we want keep only the its intensity in the source image.

I try to improve the description.